File: TheBigDay/Models/Customer.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TheBigDay.Models
{
    [Table("Customer")]
    public class Customer
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid ID { get; set; }
        public string? PhotoPath { get; set; }
        public string FirstNme { get; set; }
        public string LastName { get; set; }
        public string AddressLine1 { get; set; }
        public string? AddressLine2 { get; set; }
        public string Suburb { get; set; }
        public string State { get; set; }
        public string Country { get; set; }
        public string Postcode { get; set; }
        public string Mobile { get; set; }
        public string Email { get; set; }
        public DateTime DOB { get; set; }
        public bool IsDeleted { get; set; }

        //Dependacies
        public List<EventCustomers>? EventCustomers { get; set; }
    }
}

File: TheBigDay.Web/projects/common/src/lib/common-rest-models/customer.ts
import {Vendor} from "./vendor";

export interface Country {
    name?: string;
    code?: string;
}

export interface Representative {
    name?: string;
    image?: string;
}

export interface Customer {
    id: string
    photoPath: string;
    firstName: string;
    lastName: string;
    addressLine1: string;
    addressLine2: string;
    suburb: string;
    state: string;
    country: string;
    postcode: string;
    mobile: string;
    email: string;
    dob: Date;
    isDeleted: boolean;
    storeId?: string;
    store: Vendor | undefined;

//Dependacies
    eventCustomers: any[];
}
